frontend/docs/components/ui/cookie-banner.tsx

```diff
@@ -0,0 +1,138 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { cn } from "@/lib/utils";
+import { Button } from "./button";
+import { CookieIcon } from "@radix-ui/react-icons";
+import posthog from "posthog-js";
+
+import { loadReoScript } from "@/lib/reoWrapper";
+
+
+export function cookieConsentGiven() {
+    if (!localStorage.getItem('cookie_consent')) {
+        return 'undecided';
+    }
+    return localStorage.getItem('cookie_consent');
+}
+
+
+export default function CookieConsent({ variant = "default", demo = false, onAcceptCallback = () => { }, onDeclineCallback = () => { } }) {
+    const [isOpen, setIsOpen] = useState(false);
+    const [hide, setHide] = useState(false);
+    const [consentGiven, setConsentGiven] = useState('');
+
+    const accept = () => {
+        setIsOpen(false);
+        document.cookie = "cookieConsent=true; expires=Fri, 31 Dec 9999 23:59:59 GMT";
+
+        localStorage.setItem('cookie_consent', 'yes');
+        setConsentGiven('yes');
+
+        setTimeout(() => {
+            setHide(true);
+        }, 700);
+        onAcceptCallback();
+    };
+
+    const decline = () => {
+        setIsOpen(false);
+        localStorage.setItem('cookie_consent', 'no');
+        setConsentGiven('no');
+        setTimeout(() => {
+            setHide(true);
+        }, 700);
+        onDeclineCallback();
+    };
+
+    useEffect(() => {
+        try {
+            setIsOpen(true);
+
+            if (document.cookie.includes("cookieConsent=true") || localStorage.getItem('cookie_consent') === 'yes') {
+                accept();
+            }
+
+            if (document.cookie.includes("cookieConsent=true") || localStorage.getItem('cookie_consent') === 'no') {
+                if (!demo) {
+                    setIsOpen(false);
+                    setTimeout(() => {
+                        setHide(true);
+                    }, 700);
+                }
+            }
+        }
+        catch (e) {
+            // console.log("Error: ", e);
+        }
+    }, []);
+
+  const [reo, setReo] = useState<boolean>(false);
```

Comment:
what's `reo`?

also nit: I don't think we need the generic, and this also should probably go up above with the other state management unless it needs to be here 🤷